src/pool/pool.service.ts

```diff
@@ -0,0 +1,160 @@
+  });
+
+export const HOUR_IN_SECONDS = 60 * 60;
+export const HOURS_IN_DAY = 24;
+
+export const createQuery = (
+  blockTimestamp_lte: string | number,
+  blockTimestamp_gte: string | number,
+) => gql`
+  query Swaps {
+    swaps(
+      first: 1000,
+      where: {
+        blockTimestamp_gte: ${blockTimestamp_gte.toString()}
+        blockTimestamp_lte: ${blockTimestamp_lte.toString()}
+      }
+      orderBy: blockTimestamp
+      orderDirection: asc
+    ) {
+      id
+      tokenIn
+      tokenInAmount
+      poolId
+      tokenOut
+      tokenOutAmount
+      receiptId
+      blockTimestamp
+    }
+  }
+`;
+
+@Injectable()
+export class PoolService {
+  private readonly logger = new Logger(PoolService.name);
+
+  constructor(
+    @InjectRepository(Pool)
+    private readonly poolRepo: Repository<Pool>,
+  ) {}
+
+  @Cron('* * * * *')
+  async handleCron() {
+    this.logger.verbose('handleCron for pool service');
+
+    try {
+      const swaps = await this.requestDataForDay();
+      swaps.map((array) =>
+        this.logger.log(`Results length per hour ${array.length}`),
+      );
+      const flatSwaps = swaps.flat();
+      const newPools: { [key: string]: Pool } = flatSwaps.reduce(
+        (
+          acc: { [key: string]: Pool },
+          swap: {
+            id: string;
+            output: string;
+            poolId: number;
+            blockTimestamp: number;
+            tokenInAmount: number;
+            tokenIn: string;
+            tokenOutAmount: number;
+            tokenOut: string;
+            receiptId: string;
+            predecessorId: string;
+          },
+        ) => {
+          const [, poolId] = swap.id?.split(' ');
+          if (!poolId) return { ...acc };
+          let poolFromAcc = acc[poolId];
+
+          if (!poolFromAcc) {
+            poolFromAcc = new Pool();
+            poolFromAcc.id = poolId;
+            poolFromAcc.tokenFirst = swap.tokenIn;
+            poolFromAcc.tokenSecond = swap.tokenOut;
+            poolFromAcc.volume24hFirst = new Big(swap.tokenInAmount).toFixed(0);
+            poolFromAcc.volume24hSecond = new Big(swap.tokenOutAmount).toFixed(
+              0,
+            );
+          } else {
+            if (poolFromAcc.tokenFirst === swap.tokenIn) {
+              poolFromAcc.volume24hFirst = new Big(poolFromAcc.volume24hFirst)
+                .add(swap.tokenInAmount)
+                .toFixed(0);
+              poolFromAcc.volume24hSecond = new Big(poolFromAcc.volume24hSecond)
+                .add(swap.tokenOutAmount)
+                .toFixed(0);
+            } else {
+              poolFromAcc.volume24hFirst = new Big(poolFromAcc.volume24hFirst)
+                .add(swap.tokenOutAmount)
+                .toFixed(0);
+              poolFromAcc.volume24hSecond = new Big(poolFromAcc.volume24hSecond)
+                .add(swap.tokenInAmount)
+                .toFixed(0);
+            }
+          }
+
+          return { ...acc, [poolId]: poolFromAcc };
```

Comment:
Wouldn't be reusing acc more effective?

```typescript
acc[poolId] = poolFromAcc;
return acc;
```

This way you don't recreate an object from scratch in every iteration.

Reply:
was working with rules where mutating in reduce is strongly unrecommended)